main.go

```diff
@@ -185,7 +185,7 @@ func main() {
 	if err != nil {
 		log.Error(err, "failed to start the background process to auto-detect the operator capabilities")
 	} else {
-		autodetect.Start()
+		autodetect.DetectOpenshift()
```

Comment:
I guess we need to check for err, and repeat in some sort of loop with backoff in the event that the detection fails for some reason.  Log if there is a problem, sleep and retry. In general the env should be available, but just in case.

Reply:
will do. thx